core/shared/src/main/scala/zio/internal/FiberRuntime.scala

```diff
@@ -931,16 +937,18 @@ final class FiberRuntime[E, A](fiberId: FiberId.Runtime, fiberRefs0: FiberRefs,
     // Note that assigning `cur` as the result of `try` or `if` can cause Scalac to box local variables.
     var cur        = effect
     var done       = null.asInstanceOf[Exit[Any, Any]]
+    var cont       = true
```

Comment:
Out of curiosity, why did you end up using a flag instead of `return null`? I find that using a return statement is much more explicit on what's the method is going to return, and it's very clear on _when_ it'll return it (i.e., when we we're yielding).

Also, I doubt there is much difference in performance, but any chance we could compare them _just in case_?

Reply:
tbh, it never crossed my mind 😎 
however, like many(?) other scala devs I consider the `return` statement 'banned', I recall someone told me it was implemented using an exception and in general (at least at the time) not really recommended.

I guess I can try changing the `YieldNow` case to use return and measure

Reply:
> I recall someone told me it was implemented using an exception

That's only partially true. This happens only in cases where the `return` keyword is used within a function, e.g., `def foo: () => Int = () => return () => 1`. This is called a "non-local return", and it's indeed implemented via an exception. The Scala 3 compiler raises a warning when this happens, and in Scala 2.13 it can be made into a compiler error by using the `-Xlint:nonlocal-return` compiler flag.

In fact local returns (the "good" kind) are used extensively in the Scala Std library. E.g., the `List#contains` method is implemented as below, which is perfectly fine.

```scala
  override final def contains[A1 >: A](elem: A1): Boolean = {
    var these: List[A] = this
    while (!these.isEmpty) {
      if (these.head == elem) return true
      these = these.tail
    }
    false
  }
```

In short, the "problem" with `return` which led to it being villainised is that it breaks functional programming semantics (and performance due to non-local returns). However, in imperative-style coding (which is how the `FiberRuntime` is written to improve performance), using `return` is perfectly ok.